samples/algorithms/sudoku-grover/Sudoku.qs

```diff
@@ -143,21 +137,102 @@ namespace Microsoft.Quantum.Samples.SudokuGrover {
         }
     }
 
+
+    /// # Summary
+    /// Encodes stating number constraints into amplitudes.
+    ///
+    /// # Inputs
+    /// ## nVertices
+    /// The number of vertices in the graph.
+    /// ## bitsPerColor
+    /// The bit width for number of colors.
+    /// ## startingNumberConstraints
+    /// The array of (Vertex#, Color) specifying the disallowed colors for vertices.
+    ///
+    /// # Output
+    /// A 2D array of amplitudes where the first index is the cell and the second index is the value of a basis state (i.e., value) for the cell
+    function AllowedAmplitudes(
+        nVertices : Int,
+        bitsPerColor : Int,
+        startingNumberConstraints : (Int, Int)[]
+    ) : Double[][] {
+        mutable amplitudes = [[1.0, size=1 <<< bitsPerColor], size=nVertices];
+        for (cell, value) in startingNumberConstraints {
+            set amplitudes w/= cell <- (amplitudes[cell] w/ value <- 0.0);
+        }
+        return amplitudes;
+    }
+
+
+    /// # Summary
+    /// Prepare an equal superposition of all basis states that satisfy the constraints
+    /// imposed by the digits already placed in the grid.
+    ///
+    /// # Inputs
+    /// ## nVertices
+    /// The number of vertices in the graph.
+    /// ## bitsPerColor
+    /// The bit width for number of colors.
+    /// ## startingNumberConstraints
+    /// The array of (Vertex#, Color) specifying the disallowed colors for vertices.
+    ///
+    /// # Remarks
+    /// Creates the search space. Using the allowed amplitudes prepares uniform superposition of all allowed values for each cell
+    operation PrepareSearchStatesSuperposition(
+        nVertices : Int,
+        bitsPerColor : Int,
+        startingNumberConstraints : (Int, Int)[],
+        register : Qubit[]
+    ) : Unit is Adj + Ctl {
+        // Split the given register into nVertices chunks of size bitsPerColor.
+        let colorRegisters = Chunks(bitsPerColor, register);
+        // For each vertex, create an array of possible states we're looking at.
+        let amplitudes = AllowedAmplitudes(nVertices, bitsPerColor, startingNumberConstraints);
+        // For each vertex, prepare a superposition of its possible states on the chunk storing its color.
+        for (amps, chunk) in Zipped(amplitudes, colorRegisters) {
+            PrepareArbitraryStateD(amps, LittleEndian(chunk));
+        }
+    }
+
+    /// # Summary
+    /// Prepare equal superposition of all basis states that satisfy the constraints
```

Comment:
This summary looks like it is from the `PrepareSearchStatesSuperposition` operation rather than this function